### PR TITLE
Keep history cache dynamic config backward compatible

### PR DESCRIPTION
**Pull request by yux0**
## What changed?
Keep history cache dynamic config backward compatible

## Why?
Reuse the old dynamic config for shard level cache.

### CODE REVIEW

File: common/dynamicconfig/constants.go [hunk 1]
```diff
@@ -511,8 +511,8 @@ const (
 	HistoryCacheNonUserContextLockTimeout = "history.cacheNonUserContextLockTimeout"
 	// EnableHostHistoryCache controls if the history cache is host level
 	EnableHostHistoryCache = "history.enableHostHistoryCache"
-	// HistoryCacheShardLevelMaxSize is max size of history shard level cache
-	HistoryCacheShardLevelMaxSize = "history.shardLevelCacheMaxSize"
+	// HistoryCacheHostLevelMaxSize is max size of history host level cache
```
Comment (ast2023) [inline common/dynamicconfig/constants.go:514]: In what units do we measure history cache size?

File: service/history/configs/config.go [hunk 2]
```diff
@@ -356,11 +356,11 @@ func NewConfig(
 
 		EmitShardLagLog:                       dc.GetBoolProperty(dynamicconfig.EmitShardLagLog, false),
 		HistoryCacheInitialSize:               dc.GetIntProperty(dynamicconfig.HistoryCacheInitialSize, 128),
-		HistoryCacheMaxSize:                   dc.GetIntProperty(dynamicconfig.HistoryCacheMaxSize, 256000),
+		HistoryShardLevelCacheMaxSize:         dc.GetIntProperty(dynamicconfig.HistoryCacheMaxSize, 512),
```
Comment (ast2023) [inline service/history/configs/config.go:359]: HistoryCacheInitialSize is the size of host or shard or both?

File: common/dynamicconfig/constants.go (hunk 1 above)
Comment (yux0) [reply common/dynamicconfig/constants.go:514]: I lose track what the unit is. the config also uses during cache initialization.

File: service/history/configs/config.go (hunk 2 above)
Comment (yux0) [reply service/history/configs/config.go:359]: Not used

**Pull request merged**